pennylane/gradients/general_shift_rules.py

```diff
@@ -391,28 +391,41 @@ def generate_shifted_tapes(tape, index, shifts, multipliers=None):
             The length should match the one of ``shifts``. Each multiplier scales the
             corresponding gate parameter before the shift is applied. If not provided, the
             parameters will not be scaled.
+        broadcast (bool): Whether or not to use broadcasting to create a single tape
+            with the shifted parameters.
 
     Returns:
         list[QuantumTape]: List of quantum tapes. In each tape the parameter indicated
             by ``index`` has been shifted by the values in ``shifts``. The number of tapes
-            matches the lenth of ``shifts`` and ``multipliers`` (if provided).
+            matches the length of ``shifts`` and ``multipliers`` (if provided).
+            If ``broadcast=True`` was used, the list contains a single broadcasted tape
+            with all shifts distributed over the broadcasting dimension. In this case,
+            the ``batch_size`` of the returned tape matches the length of ``shifts``.
     """
-    params = list(tape.get_parameters())
-    if multipliers is None:
-        multipliers = np.ones_like(shifts)
 
-    tapes = []
-
-    for shift, multiplier in zip(shifts, multipliers):
+    def _copy_and_shift_params(tape, params, idx, shift, mult):
+        """Create a copy of a tape and of parameters, and set the new tape to the parameters
+        rescaled and shifted as indicated by ``idx``, ``mult`` and ``shift``."""
         new_params = params.copy()
-        shifted_tape = tape.copy(copy_operations=True)
-        new_params[index] = new_params[index] * qml.math.convert_like(multiplier, new_params[index])
-        new_params[index] = new_params[index] + qml.math.convert_like(shift, new_params[index])
+        new_params[idx] = new_params[idx] * qml.math.convert_like(
+            mult, new_params[idx]
+        ) + qml.math.convert_like(shift, new_params[idx])
 
+        shifted_tape = tape.copy(copy_operations=True)
         shifted_tape.set_parameters(new_params)
-        tapes.append(shifted_tape)
+        return shifted_tape
 
-    return tapes
+    params = list(tape.get_parameters())
+    if multipliers is None:
+        multipliers = np.ones_like(shifts)
+
+    if broadcast:
+        return [_copy_and_shift_params(tape, params, index, shifts, multipliers)]
+
+    return [
+        _copy_and_shift_params(tape, params, index, shift, multiplier)
+        for shift, multiplier in zip(shifts, multipliers)
+    ]

```

Comment:
Can these be tuples instead of lists?

Reply:
Looks like the answer is yes, from just trying it out. Collections of tapes typically are passed around as lists, should that be changed more generally? :)

Reply:
I think this would be a good idea, assuming it is not much effort. If so, we can move this into a follow-up PR.
The main advantage is the performance impact (tuples > lists), as well as the support for hashability of the entries (assuming we tackle this for the tuple contents later).